### PR TITLE
Fix null_equality config of rolling_collect_set

### LINKED ISSUE #8405: [BUG] null_equality config of rolling_collect_set doesn't work
**Describe the bug**
In current implementation of rolling_collect_set, we call `lists::detail::drop_list_duplicates` with hard-code parameters for `null_equal` and `nan_equal` by mistake, which disabled the two configs. ([code link](https://github.com/rapidsai/cudf/blob/branch-21.06/cpp/src/rolling/rolling_detail.cuh#L749))

P.S. The bug is not that urgent, since it won't do harm to the downstream projects (spark-rapids).

### PR DESCRIPTION
**Pull request by sperlingxx**
Fix #8405, and add some tests for various `null_equality` and `nan_equality`.

### CODE REVIEW

File: cpp/tests/groupby/collect_set_tests.cpp [hunk 1]
```diff
@@ -146,6 +146,47 @@ TEST_F(CollectSetTest, StringInput)
...
+  // null unequal with nan unequal
+  vals_expected = {{{-2.3e-5f, 1.0f, 2.3e5f, -NAN, -NAN, NAN, NAN, 0.0f, 0.0f},
+                    VALIDITY{true, true, true, true, true, true, true, false, false}}};
+  test_single_agg(
+    keys, vals, keys_expected, vals_expected, CollectSetTest::collect_set_null_unequal());
+  // null exclude with nan unequal
+  vals_expected = {{-2.3e-5f, 1.0f, 2.3e5f, -NAN, -NAN, NAN, NAN}};
+  test_single_agg(
+    keys, vals, keys_expected, vals_expected, CollectSetTest::collect_set_null_exclude());
+  // null equal with nan unequal
```
Comment (ttnghia) [inline cpp/tests/groupby/collect_set_tests.cpp:171]:
```suggestion
  // null equal with nan equal
```

File: cpp/tests/groupby/collect_set_tests.cpp [hunk 2]
```diff
@@ -146,6 +146,47 @@ TEST_F(CollectSetTest, StringInput)
...
+  vals_expected = {{{-2.3e-5f, 1.0f, 2.3e5f, NAN, 0.0f}, VALIDITY{true, true, true, true, false}}};
+  test_single_agg(keys,
+                  vals,
+                  keys_expected,
+                  vals_expected,
+                  cudf::make_collect_set_aggregation(
+                    null_policy::INCLUDE, null_equality::EQUAL, nan_equality::ALL_EQUAL));
+  // null unequal with nan unequal
```
Comment (ttnghia) [inline cpp/tests/groupby/collect_set_tests.cpp:179]:
```suggestion
  // null unequal with nan equal
```

**Review by ttnghia — CHANGES_REQUESTED** [review]
Also please check and fix other comments as I may miss something.

**Comment by harrism** [comment]
We are code frozen. Can this wait for 21.08?

**Review by ttnghia — APPROVED** [review]

**Review by mythrocks — APPROVED** [review]
This looks good. The only gripe I have is that the `rolling/collect_ops_test.cpp` does not specifically have a test for `nan_equality`, like you have added for `groupby`. :]

**Comment by sperlingxx** [comment]
> This looks good. The only gripe I have is that the `rolling/collect_ops_test.cpp` does not specifically have a test for `nan_equality`, like you have added for `groupby`. :]

I added it. Thanks for catching that :)

**Comment by sperlingxx** [comment]
@gpucibot merge

**Pull request merged**